include/shareddata.h

```diff
@@ -198,7 +198,8 @@ void updateDlsymOffset(int32_t dlsymOffset, int32_t dlsymOffset_m32 = 0);
 int32_t getDlsymOffset(void);
 int32_t getDlsymOffset_m32(void);
 
-int32_t getRealIPCId(int type, int32_t virt);
+int32_t getRealIPCId(int type, int32_t virt,
+                     bool insertIfNotFound = true);
```

Comment:
If I'm reading this right, the default value for 1insertIfNotFound` is 'true', and there is one invocation in which we make the argument explicit, inside `SysVShm::virtualToRealKey(key_t k)`.  But that invocation also specifies 'true' for this optional argument.

Is there any case where the optional argument would be false?  If not, then let's just hardwire the code to always insert, if not found. 

I also prefer removing the optional argument, because we eliminiate one of the non-local issues in the design of C++.  If we remove the optional arg, the reader no longer has to look at the .h file to determine if there is a missing default argument, and what is the default value.

Reply:
That's a good catch. It is supposed to be `false` to ensure we don't break any older scenarios with plugins like MANA. I have fixed it now.